e2e/testcases/resource_conditions_test.go

```diff
@@ -179,40 +175,38 @@ func TestResourceConditionAnnotations(t *testing.T) {
 		`configmanagement.gke.io/reconciling=["ClusterRole needs... something..."]`)
 
 	// Ensure reconciling conditions are added.
-	_, err1 = nomostest.Retry(waitTimeout, func() error {
+	_, err = nomostest.Retry(waitTimeout, func() error {
 		if support {
 			// We expect three reconciling conditions even though we only supplied two.
-			return nt.Validate(repo.DefaultName, "", &v1.Repo{},
+			err1 = nt.Validate(repo.DefaultName, "", &v1.Repo{},
 				hasConditions(string(v1.ResourceStateReconciling), string(v1.ResourceStateReconciling), string(v1.ResourceStateReconciling)))
-		}
-		return nt.Validate(repo.DefaultName, "", &v1.Repo{},
-			hasConditions(string(v1.ResourceStateReconciling), string(v1.ResourceStateReconciling)))
-	})
-	// The ClusterConfig condition from the ClusterRole gets duplicated.
-	// This will be obsolete with ConfigSync v2, so no need to fix (b/154226839).
-	_, err2 = nomostest.Retry(waitTimeout, func() error {
-		if support {
-			return nt.Validate(v1.ClusterConfigName, "", &v1.ClusterConfig{},
+			// The ClusterConfig condition from the ClusterRole gets duplicated.
+			// This will be obsolete with ConfigSync v2, so no need to fix (b/154226839).
+			err2 = nt.Validate(v1.ClusterConfigName, "", &v1.ClusterConfig{},
+				hasConditions(string(v1.ResourceStateReconciling), string(v1.ResourceStateReconciling)))
+		} else {
+			err1 = nt.Validate(repo.DefaultName, "", &v1.Repo{},
 				hasConditions(string(v1.ResourceStateReconciling), string(v1.ResourceStateReconciling)))
+			err2 = nt.Validate(v1.ClusterConfigName, "", &v1.ClusterConfig{},
+				hasConditions(string(v1.ResourceStateReconciling)))
 		}
-		return nt.Validate(v1.ClusterConfigName, "", &v1.ClusterConfig{},
-			hasConditions(string(v1.ResourceStateReconciling)))
-	})
-	_, err3 = nomostest.Retry(waitTimeout, func() error {
-		return nt.Validate(ns, "", &v1.NamespaceConfig{},
+		err3 = nt.Validate(ns, "", &v1.NamespaceConfig{},
 			hasConditions(string(v1.ResourceStateReconciling)))
-	})
-	if err1 != nil || err2 != nil || err3 != nil {
-		if err1 != nil {
-			nt.T.Error(err1)
-		}
-		if err2 != nil {
-			nt.T.Error(err2)
-		}
-		if err3 != nil {
-			nt.T.Error(err3)
+		if err1 != nil || err2 != nil || err3 != nil {
```

Comment:
I think this if statement is redundant. But I suspect a MultiError would be better anyway, if we're going to wait for all of them to Validate before returning.

Reply:
DONE. Switched to use MultiError.